cs/client.py

```diff
@@ -137,7 +152,7 @@ def __init__(self, endpoint, key, secret, timeout=10, method='get',
                  verify=None, cert=None, name=None, retry=0,
                  job_timeout=None, poll_interval=POLL_INTERVAL,
                  expiration=timedelta(minutes=10), trace=False,
-                 dangerous_no_tls_verify=False):
+                 dangerous_no_tls_verify=False, headers={}):
```

Comment:
Mutable datastructures as default arguments should be avoided, as they can lead to surprising behavior:

```python
def foo(headers={}):
    print("Headers", headers)
    headers.update(foo='bar')


foo()
foo()
```

Which outputs:

```
Headers {}
Headers {'foo': 'bar'}
```

Instead, the default should be None and handled in the function:

```python
def foo(…, headers=None):
    if headers is None:
        headers = {}
```

There is another occurrence below.

Reply:
Good to know, thanks: fixed.